app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListViewModel.kt

```diff
@@ -40,11 +42,19 @@ class ThreadListViewModel : ViewModel() {
     val isRecoveringFinished = MutableLiveData(true)
     val updatedAtTrigger = MutableLiveData<Unit>()
 
-    val currentFolder = SingleLiveEvent<Folder?>(null)
+    val currentFolder = Transformations.switchMap(MainViewModel.currentFolderId) { folderId ->
+        liveData(Dispatchers.IO) {
+            if (folderId != null) emitSource(FolderController.getFolderAsync(folderId).mapNotNull { it.obj }.asLiveData())
+        }
+    }
+
     val currentFilter = SingleLiveEvent(ThreadFilter.ALL)
+
     val currentThreads = Transformations.switchMap(observeFolderAndFilter()) { (folder, filter) ->
         liveData(Dispatchers.IO) {
-            if (folder != null) emitSource(ThreadController.getThreads(folder.id, filter).asFlow().asLiveData())
+            if (folder != null && folder.isValid()) {
```

Comment:
When we switch to another Mailbox, it crashes here because it tries to access to the Folders of the previous Mailbox.
But since we closed the Realm of this previous Mailbox, it can't access it, hence the crash.